lib/info.js

```diff
@@ -43,6 +43,14 @@ const AGE_RESTRICTED_URLS = [
 */
 exports.getBasicInfo = async(id, options) => {
   const retryOptions = Object.assign({}, miniget.defaultOptions, options.requestOptions);
+  if (options.requestOptions === undefined) {
+    options.requestOptions = {};
+  }
```

Comment:
we should try not to alter any objects passed to the function. this will add a `requestOptions` to the original `options`, which may confuse users who later see an extra field. we can instead overwrite the `options` given with one that contains a default `requestOptions`
```suggestion
options = Object.assign({}, options, {
  requestOptions: {},
});
```

Reply:
this actually breaks a couple of tests, I got it working with:
```javascript
options.requestOptions = Object.assign({}, options.requestOptions, {});
```